scripts/skills/perks/perk_rf_fresh_and_furious.nut

```diff
@@ -67,24 +67,16 @@ this.perk_rf_fresh_and_furious <- ::inherit("scripts/skills/skill", {
 
 	function onAfterUpdate( _properties )
 	{
-		if (!this.m.IsSpent && !this.m.RequiresRecover)
-		{
-			foreach (skill in this.getContainer().getAllSkillsOfType(::Const.SkillType.Active))
-			{
-				// ::Math.round to round up the subtraction because we want to emulate the behavior of _properties.IsSkillUseHalfCost
-				// whereby it rounds down the cost (due to integer division) after halving it.
-				skill.m.ActionPointCost -= ::Math.max(0, ::Math.min(skill.m.ActionPointCost - 1, ::Math.round(skill.m.ActionPointCost / 2.0)));
-			}
-		}
-	}
+		if (this.m.IsSpent || this.m.RequiresRecover)
+			return;
 
-	function onAffordablePreview( _skill, _movementTile )
-	{
-		if (_skill != null && _skill.getActionPointCost() != 0 && _skill.getFatigueCost() != 0)
+		local actor = this.getContainer().getActor();
+		if (!actor.isPreviewing() || actor.getPreviewMovement() != null || (actor.getPreviewSkill().getActionPointCost() == 0 && actor.getPreviewSkill().getFatigueCost() == 0))
```

Comment:
As discussed on voice chat, I prefer brackets around `&&` separated expressions inside `if` blocks because they vastly improve readability of such statements i.e. the `||` separated blocks are more clearer this way. Taro likes this too. So this is one of those cases where adding brackets is fine.

Reply:
Is this a hard or soft rule?
This should probably be written down somewhere. Only thing I remember and have for reference is the Enduriel Statement where he said that cases where you feel the need to use optional brackets indicate that the whole if condition should be split up and restructured.

Reply:
There's no rule for this, it's just how I do it. What Enduriel said is correct but it's a general statement and that's correct but we can't really make that into a rule as it varies from situation to situation. The brackets in mixed `||` and &&` are good to have if we aren't splitting them up.